io/pci/EEH.py

```diff
@@ -63,7 +64,22 @@ def setUp(self):
         eeh_enable_file = "/sys/kernel/debug/powerpc/eeh_enable"
         if '0x1' not in genio.read_file(eeh_enable_file).strip():
             self.cancel("EEH is not enabled, please enable via FSP")
-        self.max_freeze = self.params.get('max_freeze', default=1)
+        self.max_freeze = self.params.get('max_freeze')
+        self.log.info('Reading max supported freeze from file eeh_max_freezes')
+        self.max_supported_freeze = int(process.system_output(
+            'cat /sys/kernel/debug/powerpc/eeh_max_freezes'))
+        if self.max_freeze is None:
+            self.max_freeze = self.max_supported_freeze
+            dmesg_op = process.system_output('dmesg',
+                                             ignore_status=True,
+                                             shell=True).decode("utf-8")
+            if 'times in the last hour' in dmesg_op:
+                eeh_already_hit = re.findall(r"This PCI device has failed "
+                                             "([0-9]+) times in the last hour and will be "
+                                             "permanently disabled after ([0-9]+) failures", dmesg_op)
+                self.log.info('max_freeze value is adjusted as %s eeh hits'
+                              ' already observed', eeh_already_hit[-1][0])
+                self.max_freeze -= int(eeh_already_hit[-1][0])
```

Comment:
If the adapter already disabled due to max hits. than why adjusting value.. or if eeh not yet trigger there is no point checking dmesg .. am I missing something here ?

Reply:
Will be addressing it in PR "https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2590"